staging/src/k8s.io/apiextensions-apiserver/pkg/controller/openapi/builder/builder.go

```diff
@@ -486,13 +528,14 @@ func (b *builder) getOpenAPIConfig() *common.Config {
 		},
 		GetOperationIDAndTags: openapi.GetOperationIDAndTags,
 		GetDefinitionName: func(name string) (string, spec.Extensions) {
-			buildDefinitions.Do(buildDefinitionsFunc)
+			buildDefinitions.Do(generateBuildDefinitionsFunc(v2))
 			return namer.GetDefinitionName(name)
 		},
 		GetDefinitions: func(ref common.ReferenceCallback) map[string]common.OpenAPIDefinition {
 			def := generatedopenapi.GetOpenAPIDefinitions(ref)
 			def[fmt.Sprintf("%s/%s.%s", b.group, b.version, b.kind)] = common.OpenAPIDefinition{
-				Schema: *b.schema,
+				Schema:       *b.schema,
+				Dependencies: []string{objectMetaType},
```

Comment:
We used to serve an incomplete crd schema without the objectmeta reference, but it was never surfaced because the aggregator would include the objectmeta in during aggregation.